### ISSUE TITLE
Mapping buttons to axes

### ISSUE THREAD

**Issue by delallea** [evilC/AHK-Universal-Joystick-Remapper#15]
I guess it's not currently possible (at least I haven't found how), but I think it would be a useful feature: make it possible to map a physical button to a virtual axis.

My use case is for a game that uses the Z axis, but my controller doesn't provide one (it only has X/Y and RX/RY). However there are buttons I could use (obviously I wouldn't be able to have a fine grained control on the axis values, but mapping the buttons to axis boundaries would be enough).

**Comment by evilC** [comment]
Not possible with the current implementation of UJR, I am afraid.
This is on the feature list for the next iteration of UJR though, but that could be months or even years off.
If you are able to code AHK scripts, what you seek would not be too hard. You would want my CvJoyInterface library to allow you to control the vJoy stick from AHK.

**Comment by delallea** [comment]
Ok thanks. I ended up doing exactly what you suggest :)
Feel free to close this issue unless you want to keep it around for tracking purpose.

**Comment by My1** [comment]
while we are at it, it would also be epic to convert a d-pad to axes and buttons to a d-pad. that would make near-perfect joycon usage on PC possible.

**Comment by evilC** [comment]
This is possible using my new app, [UCR](https://autohotkey.com/boards/viewtopic.php?f=19&t=12249).  
UJR is no longer maintained...

**Comment by My1** [comment]
okay didnt see taht because I just got to the issues page with the link from UJR.

might be helpful to add a link to the new version to the page of UJR
http://evilc.com/joomla/articles/9-code/autohotkey/1-ujr-universal-joystick-remapper

**Comment by evilC** [comment]
Yeah, I am just starting to redirect UJR users to UCR. I have a redirect on the forum thread for UJR, but you are right, I should put one on the front page of the UJR repo, and on my website.

**Comment by My1** [comment]
I know the tool from half an eternity ago and just found the tutorial.

**Comment by evilC** [comment]
I am also thinking of putting something in the download ZIP for UJR, or even release a "new" version of UJR so people see "updates available", but the new version tells them about UCR.  
UJR still has more than 4x the number of downloads per month than UCR, so clearly there are many people who do not know.  
I do not want to break UJR though or introduce nag screens etc, as UCR is still in Alpha stage and UJR has stood the test of time.
